### ISSUE TITLE
Smartcase support?

### ISSUE THREAD

**Issue by endrebak** [hlissner/evil-snipe#9]
Thanks for a great package. Evil-snipe is a must-have.

One thing I am missing though is smartcase, which is a brilliant vim feature.

It is simply 
```
if a pattern contains an uppercase letter, it is case sensitive, otherwise, it is not. 
For example, /The would find only "The", while /the would find "the" or "The" etc. 
```

It would be nice to have smartcase searches with s (as an option at least). `f` and `t` are probably better cased.

**Comment by hlissner** [comment]
Good idea! I've implemented it as of 1.5.8 with the option `evil-snipe-smart-case`:

`(setq evil-snipe-smart-case t)`

I haven't tested it extensively, so let me know if there are any problems.

**Comment by endrebak** [comment]
Seems to be working brilliantly, but will report any bugs/glitches.
